app/connector/FrontendAuthConnector.scala

```diff
@@ -31,14 +35,43 @@ object GovernmentGatewayEnrolment {
   implicit val fmt = Json.format[GovernmentGatewayEnrolment]
 }
 
+case class InternalUserIdentifier(value: String) extends AnyVal
+
+object InternalUserIdentifier {
+  implicit val reads: Reads[InternalUserIdentifier] = new Reads[InternalUserIdentifier] {
+    override def reads(json: JsValue): JsResult[InternalUserIdentifier] =
+      Try((json \ "internalId").as[String])
+        .map(id => JsSuccess(InternalUserIdentifier(id)))
+        .getOrElse(JsError("Error parsing internal user identifier"))
+  }
+
+  implicit def toString(id: InternalUserIdentifier): String = id.value
```

Comment:
I see your point and I think I would convert this to be `convertToString`. What do you think?